pkg/k8shandler/elasticsearch.go

```diff
@@ -434,3 +434,76 @@ func (req *ElasticsearchRequest) ListIndicesForAlias(aliasPattern string) ([]str
 	}
 	return response, nil
 }
+
+func AddAliasForOldIndices(clusterName, namespace string, client client.Client) (bool, error) {
+	// get .operations.*/_alias
+	// get project.*/_alias
+	/*
+		{
+			"project.test.107d38b1-413b-11ea-a2cd-0a3ee645943a.2020.01.27" : {
+				"aliases" : {
+					"test" : { }
+				}
+			},
+			"project.test2.8fe8b95e-4147-11ea-91e1-062a8c33f2ae.2020.01.27" : {
+				"aliases" : { }
+			}
+		}
+	*/
+
+	successful := true
+
+	payload := &esCurlStruct{
+		Method: http.MethodGet,
+		URI:    "project.*,.operations.*/_alias",
+	}
+
+	curlESService(clusterName, namespace, payload, client)
+
+	// alias name choice based on https://github.com/openshift/enhancements/blob/master/enhancements/cluster-logging/cluster-logging-es-rollover-data-design.md#data-model
+	for index := range payload.ResponseBody {
+		// iterate over each index, if they have no aliases that match the new format
+		// then PUT the alias
+
+		indexAlias := ""
+		if strings.HasPrefix(index, "project.") {
+			// it is a container log index
+			indexAlias = "app"
+		} else {
+			// it is an operations index
+			indexAlias = "infra"
+		}
+
+		if payload.ResponseBody[index] != nil {
+			indexBody := payload.ResponseBody[index].(map[string]interface{})
+			if indexBody["aliases"] != nil {
```

Comment:
What happens if it is nil?  Don't we need to still alias the index?

Reply:
if its nil then we didn't get a complete response back from ES, not something we expect but its necessary since we have `interface{}` objects to parse

Reply:
I'm not really asking why we are checking for nil more that regardless if nil or not, we still want to ensure an index has the alias we desire.  Should we be aliasing whether it's nil or not or is it ok to only do this work if its not nil.

Reply:
i think it would be all right to not alias if this is null. we would end up trying again at a later time. in my mind, if we're getting a response from ES that has this null that means it isn't properly responding to queries and likely isn't ready to accept a request to create an alias.